c7n/resources/ecr.py

```diff
@@ -70,25 +68,16 @@ class ECRRemoveTags(tags.RemoveTag):
 
     permissions = ('ecr:UntagResource',)
 
-    def process_resource_set(self, resources, tags):
-        client = local_session(
-            self.manager.session_factory).client('ecr')
+    def process_resource_set(self, client, resources, tags):
         for r in resources:
             try:
                 client.untag_resource(resourceArn=r['repositoryArn'], tagKeys=tags)
             except client.exceptions.RepositoryNotFoundException:
                 pass
 
 
-@ECR.action_registry.register('mark-for-op')
```

Comment:
? just removing mark-for-op for ECR?

Reply:
a few lines further down you'll see
```
ECR.action_registry.register('mark-for-op', tags.TagDelayedAction)
```

tags.TagDelayedAction is generic at this point, for extant subclasses with good doc strings examples that were resource specific, I left the extant subclass registrations but removed the implementations. For ecr there wasn't any value add subclass docstring, so I removed the subclass and registered with directly with the base class.